core/base/src/TApplication.cxx

```diff
@@ -628,20 +630,453 @@ TApplication::EExitOnException TApplication::ExitOnException(TApplication::EExit
+      TString cLinux("xdg-open ");
+      // For Linux we check if the DISPLAY is set.
+      //TString displayHandle(gSystem->Getenv("DISPLAY"));
+      //if ((displayHandle.Sizeof()) > 1) {
+      if (gSystem->Getenv("DISPLAY")) {
+         // If the DISPLAY is set it will open the browser.
+         cLinux.Append(url);
+         gSystem->Exec(cLinux);
+      } else {
+         // Else the user will have a warning and the URL in the terminal.
+         Warning("OpenInBrowser", "The DISPLAY is not set!");
+         std::cout << url.Data() << std::endl;
+      }
+      #endif
+}
+
+enum EUrl {kURLforClass, kURLforNameSpace};
+namespace {
 ////////////////////////////////////////////////////////////////////////////////
-/// Print help on interpreter.
+/// The function generates a URL address for class or namespace (name).
+/// This is the URL to the online reference guide, generated by Doxygen.
+/// With the enumeration "EUrl" we pick which case we need - the one for
+/// class (kURLforClass) or the one for namespace (kURLforNameSpace).
+///
+/// \param[in] name <the name of the class or the namespace
+/// \param[in] type <the enumerator for class or namespace
 
-void TApplication::Help(const char *line)
+static TString UrlGenerator(const TString &name, EUrl type)
 {
-   gInterpreter->ProcessLine(line);
+   TString url("");
+   if (type == kURLforClass){
+      url.Append("https://root.cern.ch/doc/master/class");
+   } else if (type == kURLforNameSpace) {
+      url.Append("https://root.cern/doc/master/namespace");
+   }
+   url.Append(name);
+   url.Append(".html");
+   return url;
+}
+}
+
+namespace {
+////////////////////////////////////////////////////////////////////////////////
+/// The function returns a TString with the arguments of a member function
+/// (methodName) of a class (className), but modified with respect to Doxygen
+/// syntax - spacing around special sumbols and adding the missing sumbols
+/// ("std::", ",..."). "FormatMethodForDoxygen" works for the functions defined
+/// inside namespaces as well. We pass pointer to the class as well, to make
+/// sure the class exists.
+///
+/// \param[in] className <the name of the class
+/// \param[in] classVariable <pointer to the class
+/// \param[in] methodName <the name of the method from the class
 
-   Printf("\nROOT special commands.");
-   Printf("===========================================================================");
-   Printf("             pwd          : show current directory, pad and style");
-   Printf("             ls           : list contents of current directory");
-   Printf("             which [file] : shows path of macro file");
+static TString FormatMethodForDoxygen(const TString &className, TClass *classVariable, const TString &methodName)
+{
+   // With "GetSignature" we get the arguments of "methodName" and put them in a TString.
+   TString methodArguments(((TFunction *)(classVariable->GetListOfMethods()->FindObject(methodName)))->GetSignature());
```

Comment:
What if `methodName` is a global function? I'd expect `classVariable` to be null in that case, and the expression `classVariable->GetListOfMethods()` to crash. If `classVariable` is null you need to use `gROOT->GetListOfFunctions()`